beacon-chain/operations/blstoexec/pool.go

```diff
@@ -155,6 +169,9 @@ func (p *Pool) MarkIncluded(change *ethpb.SignedBLSToExecutionChange) error {
 
 	delete(p.m, change.Message.ValidatorIndex)
 	p.pending.Remove(node)
+	if p.numPending() == blsChangesPoolThreshold {
```

Comment:
Couldn't we see a situation where the size of the pool fluctuates around the value of `blsChangesPoolThreshold`, causing us to trigger this expensive copy over and over? I assumed you would only cycle if the map gone through a certain number of key deletions (and was under a max size to keep the cost of cycling down).